### PR TITLE
Fix missing data warning.

### PR DESCRIPTION
**Pull request by trivialfis**
Not critical as the missing field is converted to nan in `DMatrix.__init__`.  Just fixing the correctness of the helper function and making it self-contained.

### CODE REVIEW

File: tests/python/test_dmatrix.py [hunk 1]
```diff
@@ -12,6 +13,29 @@
...
+    def test_warn_missing(self):
+        from xgboost import data
+        with pytest.warns(UserWarning):
+            data._warn_unused_missing('uri', 4)
+
+        with pytest.warns(None) as record:
+            data._warn_unused_missing('uri', None)
+            data._warn_unused_missing('uri', np.nan)
+
+        assert len(record) == 0
```
Comment (hcho3) [inline tests/python/test_dmatrix.py:25]: Which `record` is this referring to?

Comment (trivialfis) [reply tests/python/test_dmatrix.py:25]: @hcho3 At line 21.

Comment (hcho3) [reply tests/python/test_dmatrix.py:25]: Is it legal to refer to a `with` variable past its scope? It's as if:
```
with open(..) as f:
   ...
print('foobar', file=f)
```
which seems undefined.

Comment (trivialfis) [reply tests/python/test_dmatrix.py:25]: Let me move that inside.

Comment (trivialfis) [reply tests/python/test_dmatrix.py]: Done.

**Comment by trivialfis** [comment]
@hcho3 Thanks for the speedy review.

**Pull request merged**